Source/DataSources/GeometryVisualizer.js

```diff
@@ -142,23 +142,31 @@ define([
         this._changedObjects = new AssociativeArray();
 
         var numberOfShadowModes = ShadowMode.NUMBER_OF_SHADOW_MODES;
-        this._outlineBatches = new Array(numberOfShadowModes);
-        this._closedColorBatches = new Array(numberOfShadowModes);
-        this._closedMaterialBatches = new Array(numberOfShadowModes);
-        this._openColorBatches = new Array(numberOfShadowModes);
-        this._openMaterialBatches = new Array(numberOfShadowModes);
+        this._outlineBatches = new Array(numberOfShadowModes*2);
```

Comment:
Ugh, are we doubling the number of all batches here?

Reply:
Yes, because not all geometry types have support for the `applyOffset` vertex attribute required for the `GeometryOffsetInstanceAttribute`.  If we didn't want to double the size of the batches I suppose I could add an `applyOffset` option to `BoxGeometry`, `CylinderGeometry` and all the others.  This seemed like a better solution to me.

Reply:
There's only one GeometryVisualizer per-datasource these days, right?  If so then I suppose this is OK for now.

Reply:
Although, ultimately wouldn't it make sense to have the offset and clamping capabilities to these other geometries as well?

Reply:
Yes, only one `GeometryVisualizer` per datasource.
And yes, this should let us do clamping with the other geometry types but the solution would be a bit difference since this relies on setting `height` and/or `extrudedHeight`